testrunner/extract_test.go

```diff
@@ -116,7 +117,6 @@ func TestExtractTestCode(t *testing.T) {
 		card string
 		want int
 	}{
-	
```

Comment:
General comment, not specific to this line:
You added/removed some empty lines so that the line count works out but the problem currently is that if I put in this code
```
func TestParseCard(t *testing.T) {
	tt := struct {
		name string
		card string
		want int
	}{

		name: "parse jack",
		card: "jack",
		want: 10,
	}
	
	if got := ParseCard(tt.card); got != tt.want {
		t.Errorf("ParseCard(%s) = %d, want %d", tt.card, got, tt.want)
	}
}
```
the test also passes.

Is there a way to improve the test so that we know the empty lines are actually in the places we expect them to be?

Reply:
The empty line I added (line 108) to make the line counts match, is definitely a hack. There is another empty line on line 128 that was already there and plays the same role. I tried for some time to remove these empty lines from the extracted source code, but couldn't.

I think that instead of counting line numbers, we could compare the source code and extracted code line by line, but only care whether the corresponding lines are either both empty(only contain whitespace) or both "filled"(contain non-whitespace characters). This way we ensure that not only the expected source code and extracted code have the same number of lines, but also that the empty lines are in the same spots.
If that sounds good, I'll try to update the tests.

Reply:
Instead of just checking filled vs. not filled, couldn't we restructure the existing test code to go through actual and expected code line by line and check whether they are the same when leading and trailing whitespace was removed?

Reply:
Yes, that would be even better. I'll go ahead and do that.